airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/write/load/SnowflakeInsertBuffer.kt

```diff
@@ -60,56 +43,54 @@ class SnowflakeInsertBuffer(
         }
 
     fun accumulate(recordFields: Map<String, AirbyteValue>) {
-        if (csvFilePath == null) {
-            val csvFile = createCsvFile()
-            csvFilePath = csvFile.toPath()
-            csvWriter = csvWriterBuilder.build(GZIPOutputStream(csvFile.outputStream()))
+        if (buffer == null) {
+            buffer = ByteArrayOutputStream()
         }
-
-        writeToCsvFile(recordFields)
+        bufferCsvRecord(recordFields)
     }
 
-    suspend fun flush() {
-        csvFilePath?.let { filePath ->
+    suspend fun flush() =
+        buffer?.let { b ->
             try {
-                // Flush and close the CSV write to ensure that any pending writes are written
-                // to the file AND that any proper end of file markers are written by the close
-                csvWriter?.flush()
-                csvWriter?.close()
                 logger.info { "Beginning insert into ${tableName.toPrettyString(quote = QUOTE)}" }
-                // Next, put the CSV file into the staging table
-                snowflakeClient.putInStage(tableName, filePath.pathString)
-                // Finally, copy the data from the staging table to the final table
-                snowflakeClient.copyFromStage(tableName, filePath.fileName.toString())
+                val fileName = "snowflake${java.lang.Long.toUnsignedString(random.nextLong())}.csv.gz"
+                buffer?.flush()
+                val inputStream = ByteArrayInputStream(b.toByteArray())
```

Comment:
You could try what I did [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-clickhouse/src/main/kotlin/io/airbyte/integrations/destination/clickhouse/write/load/BinaryRowInsertBuffer.kt#L100) to avoid the copy